evm/src/cpu/kernel/interpreter.rs

```diff
@@ -809,6 +918,70 @@ impl<'a> Interpreter<'a> {
     }
 }
 
+fn signed_cmp(x: U256, y: U256) -> Ordering {
+    // Signed integers do not have a unique zero representation.
+    let x_is_zero = x.is_zero() || x.eq(&MINUS_ZERO);
+    let y_is_zero = y.is_zero() || y.eq(&MINUS_ZERO);
+
+    if x_is_zero && y_is_zero {
+        return Ordering::Equal;
+    }
+
+    let x_is_pos = x.eq(&(x & SIGN_MASK));
+    let y_is_pos = y.eq(&(y & SIGN_MASK));
+
+    if x_is_zero {
+        if y_is_pos {
+            return Ordering::Less;
+        } else {
+            return Ordering::Greater;
+        }
+    };
+
+    if y_is_zero {
+        if x_is_pos {
+            return Ordering::Greater;
+        } else {
+            return Ordering::Less;
+        }
+    };
+
+    match (x_is_pos, y_is_pos) {
+        (true, true) => x.cmp(&y),
+        (true, false) => Ordering::Greater,
+        (false, true) => Ordering::Less,
+        (false, false) => x.cmp(&y).reverse(),
+    }
+}
+
+const MINUS_ZERO: U256 = U256([
```

Comment:
Hm the yellowpaper mentions two's complement. If I'm not mistaken, only the all-zero-bits encoding is interpreted as zero, while this value would be interpreted as 2^-255 (which is special in that there's no positive 2^255).

Reply:
Ah mb. I'll adapt the methods taking into account the two's complements.

Reply:
I should have adapted all methods with two's complement!

Reply:
Should be good for review!

Reply:
I'll also add some tests for all those signed arith. operations in the related `tests` module, for consistency.